gpus/example2/example2.cu

```diff
@@ -0,0 +1,89 @@
+#include <vector>
+#include <algorithm>
+#include <numeric>
+#include <iostream>
+#include <execution>
+#include <thrust/host_vector.h>
+#include <thrust/device_vector.h>
+#include <chrono>
+#include <thrust/transform_reduce.h>
+#include <thrust/execution_policy.h>
+#include <cuda/std/functional>
+#include <thrust/iterator/zip_iterator.h>
+#include <thrust/tuple.h>
+
+// dot product
+// we do sum a_i * b_i
+
+template<typename T>
+using vector = thrust::host_vector<T>;
+
+int main() {
```

Comment:
Adding comments to this file to help readers follow along and verify what different parts of the code are doing would be helpful